### PR TITLE
Ruby 3 compat

### LINKED ISSUE #190: Ruby 3.0 support for broadcast with kwargs
When broadcasting with kwargs, an `ArgumentError` is raised in Ruby 3.0:

```ruby
broadcast(:user_changed, initial_id: user_id_before_last_save, final_id: user_id)
```

```ruby
ArgumentError: wrong number of arguments (given 1, expected 0; required keywords: initial_id, final_id)
```

This worked fine on Ruby 2.6, but because `Publisher#broadcast` doesn't handle kwargs, it breaks in 3.0.

### PR DESCRIPTION
**Pull request by doits**
Fixes #190

Makes use of `**kwargs` everywhere so it is compatible with ruby 3.

Unfortunately this does not work with ruby 2.6 and below, so I dropped it. It could be made to work with 2.6 by using [ruby2_keywords](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/#a-compatible-delegation), but I assumend support for 2.6 is not a priority for this gem.

Another thing is I don't know how this works with `jruby`, e.g. if ruby mri 3 and the use of `**kword` is compatible to jruby or not.

I hope I didn't miss a place to convert to kword args.

### CODE REVIEW

**Comment by doits** [comment]
Feel free to review this now, I'll test it for myself for some time to see if I find something more.

File: wisper.gemspec [hunk 1]
```diff
@@ -17,6 +17,8 @@ Gem::Specification.new do |gem|
   gem.homepage      = "https://github.com/krisleech/wisper"
   gem.license       = "MIT"
 
+  gem.required_ruby_version = '>= 2.7'
```
Comment (krisleech) [inline wisper.gemspec:20]: Is this strictly needed as it would mean a major version bump as it would be backwards incompatible.

Comment (jcavalieri) [reply wisper.gemspec:20]: I was wondering this too. I thought the double splat is backwards compatible. @doits have you tried your MR with a lower version of Ruby?

Comment (jcavalieri) [reply wisper.gemspec:20]: https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

Unless I'm missing something it should be backwards compatible.

Comment (doits) [reply wisper.gemspec:20]: For ruby 2.6 you have to use `ruby2_keywords`, see https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/#a-compatible-delegation

This PR could be rewritten to use `ruby2_keywords`, then it should be compatible with ruby 2.6, too. Then, after 2.6 EOL, it must be rewritten to remove `ruby2_keywords` again (like it is done here).

**Comment by anisharamnani** [comment]
hi there! i was just wondering when/if this will be merged in. if not, is there anything else missing here? i'd like to upgrade to ruby 3 and continue to use this gem. thanks!

**Comment by doits** [comment]
I'm using my branch with ruby 3.1 project and have not encountered any issues yet. Hopefully this will be reviewed/merged and released soon.

**Comment by krisleech** [comment]
Thanks for your work on this.

Have you been running this in production using 3.x?

Do we know this still works with Ruby 2.6 and 2.7 in production?

What are you thoughts on merging this PR, versus releasing without 2.6 support (i.e. `ruby2_keywords`), which would not be backwards compatible and require a major version bump.

**Comment by doits** [comment]
Yes, I'm using it in production with ruby 3.1 since end of last year. Before  I used it with ruby 2.7 since I opened this PR. I use it mostly like below, so probably not all edge cases are tested in production:

```rb
  Wisper.subscribe(
    EventApplicationListener::MoveUpWaitingList,
    scope: :EventApplication,
    on: %i[state_changed],
    prefix: :on
  )
``` 

I have no idea about ruby 2.6 in production though, but it should work if you trusts the specs ;-)

Dropping support for 2.6 is cleaner (since no `ruby2_keywords`). You compare see the changes to master here:

- with ruby 2.6 support: https://github.com/krisleech/wisper/compare/master...doits:832a22cf576831163638c1ed4c0a81147e4d5e3b
- without ruby 2.6 support: https://github.com/krisleech/wisper/compare/master...doits:b86c57d6064495ef67391405e517dbe85b622d8c

**Comment by krisleech** [comment]
Let's drop support for 2.6 and I'll bump the major version. Would you mind updating this PR to remove the commit which adds `ruby2_keywords` and we can do a release.

Can you also add an item to the unreleased section of: https://github.com/krisleech/wisper/blob/master/CHANGELOG.md#head-unreleased

👍

**Comment by oskarpearson** [comment]
Can probably be closed given the work that's going into https://github.com/krisleech/wisper/pull/194 ?

**Pull request closed without merging**